metadata/dashboard/dashboard_metadata.go

```diff
@@ -967,6 +972,96 @@ func (m *MetadataServer) GetVersionMap(c *gin.Context) {
 	c.JSON(200, versionMap)
 }
 
+type SourceDataResponse struct {
+	Columns []string   `json:"columns"`
+	Rows    [][]string `json:"rows"`
+}
+
+func (m *MetadataServer) GetSourceData(c *gin.Context) {
+	name := c.Query("name")
+	variant := c.Query("variant")
+	var limit int64 = 50
+	response := SourceDataResponse{}
+	if name == "" || variant == "" {
+		fetchError := &FetchError{StatusCode: 500, Type: "GetSourceData - Could not find the name or variant query parameters"}
+		m.logger.Errorw(fetchError.Error(), "Metadata error")
+		c.JSON(fetchError.StatusCode, fetchError.Error())
+		return
+	}
+	iter, err := m.getSourceDataIterator(name, variant, limit)
+	if err != nil {
+		fetchError := &FetchError{StatusCode: 500, Type: "GetSourceData - getSourceDataIterator() threw an exception"}
+		m.logger.Errorw(fetchError.Error(), "Metadata error", err)
+		c.JSON(fetchError.StatusCode, fetchError.Error())
+		return
+	}
+	for _, columnName := range iter.Columns() {
+		response.Columns = append(response.Columns, strings.ReplaceAll(columnName, "\"", ""))
+	}
+	for iter.Next() {
+		sRow, err := serving.SerializedSourceRow(iter.Values())
+		if err != nil {
+			fetchError := &FetchError{StatusCode: 500, Type: "GetSourceData"}
+			m.logger.Errorw(fetchError.Error(), "Metadata error", err)
+			c.JSON(fetchError.StatusCode, fetchError.Error())
+			return
+		}
+		dataRow := []string{}
+		for _, currentRow := range sRow.Rows {
+			split := strings.Split(currentRow.String(), ":")
```

Comment:
Can put these operations into their own function with a name that makes it more clear what its doing